runtime/src/comm/ofi/comm-ofi.c

```diff
@@ -1906,7 +1909,14 @@ struct fi_info* getBaseProviderHints(chpl_bool* pTxAttrsForced) {
                                  | FI_MR_PROV_KEY // TODO: avoid pkey bcast?
                                  | FI_MR_ENDPOINT);
 
-  if (chpl_numNodes > 1 && envMaxHeapSize != 0 && !chpl_get_oversubscribed()) {
+  // Set FI_MR_ALLOCATED if there is more than one node and the maximimum
+  // heap size was specified and the CHPL_RT_OVERSUBSCRIBED environment
+  // variable was not set, otherwise we risk running out of memory when
+  // the heap is allocated. If CHPL_RT_OVERSUBSCRIBED is set then we may
+  // be sharing the node with another process, in which case we can't size
```

Comment:
Just a nit, but "other processes" might be more accurate than "another process" which implies merely 1.